pygmt/helpers/utils.py

```diff
@@ -12,74 +12,78 @@
 import warnings
 import webbrowser
 from collections.abc import Iterable, Sequence
-from typing import Any
+from typing import Any, Literal
 
 import xarray as xr
 from pygmt.encodings import charset
 from pygmt.exceptions import GMTInvalidInput
 
 
-def _validate_data_input(
+def validate_data_input(
```

Comment:
Make the function public, because now it's called in `Session.virtualfile_in`.

Reply:
Mentioned at https://github.com/GenericMappingTools/pygmt/pull/2744#discussion_r1435898105. I'm still unsure if we should make this public since this isn't something regular users should use. We can still import `_validate_data_input` and use it in `Session.virtualfile_in` no?

Reply:
OK, changed it back to private in 7ff0d9e2698bf23339a3c271129e63b827245d4b.